### PR TITLE
JAMES-2441 Ensure mails are well received before first connection when using a LDAP

### PR DESCRIPTION
**Pull request by chibenwa**
Also play again `cassandra-guice-ldap` tests!

### CODE REVIEW

File: server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepository.java [hunk 1]
```diff
@@ -462,21 +350,21 @@ public Context newDelegate() throws NamingException {
     protected Properties getContextEnvironment() {
         final Properties props = new Properties();
         props.put(Context.INITIAL_CONTEXT_FACTORY, INITIAL_CONTEXT_FACTORY);
-        props.put(Context.PROVIDER_URL, null == ldapHost ? "" : ldapHost);
-        if (null == credentials || credentials.isEmpty()) {
+        props.put(Context.PROVIDER_URL, null == ldapConfiguration.getLdapHost() ? "" : ldapConfiguration.getLdapHost());
```
Comment (mbaechler) [inline server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepository.java]: could we replace null checks by Optional wrapping while we are at it ?

Comment (chibenwa) [reply server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepository.java]: Sure!

I did not wanted to do complex refactoring here to be sure to not mess up...

**Comment by chibenwa** [comment]
**mvn tets** are ok.

Can be considered green.

Jenkins no node found...

test this please anyway!

**Comment by chibenwa** [comment]
test this please anyway

**Comment by aduprat** [comment]
Build triggered. sha1 is merged.

**Comment by aduprat** [comment]
Build started sha1 is merged.

**Comment by aduprat** [comment]
Build finished. No test results found.

**Comment by aduprat** [comment]
Build triggered. sha1 is merged.

**Comment by aduprat** [comment]
Build started sha1 is merged.

**Comment by aduprat** [comment]
Build finished. No test results found.

**Comment by aduprat** [comment]
Build triggered. sha1 is merged.

**Comment by aduprat** [comment]
Build started sha1 is merged.

**Comment by aduprat** [comment]
Build finished. No test results found.

**Comment by aduprat** [comment]
Build triggered. sha1 is merged.

**Comment by aduprat** [comment]
Build started sha1 is merged.

**Comment by aduprat** [comment]
Build finished. No test results found.

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[d0b640383efcf7b6dee866180105f7f3d805029d] Tests in error: 
[d0b640383efcf7b6dee866180105f7f3d805029d]   ToRepositoryTest.userShouldBeAbleToAccessReprocessedMails:133 » ConditionTimeout
```

test this please

**Comment by aduprat** [comment]
test this please

File: server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJamesServerTest.java [hunk 2]
```diff
@@ -19,21 +19,37 @@
...
     private static final String JAMES_USER = "james-user";
     private static final String PASSWORD = "secret";
     private static final String DOMAIN = "james.org";
     private static final String ADMIN_PASSWORD = "mysecretpassword";
+    private static Duration slowPacedPollInterval = FIVE_HUNDRED_MILLISECONDS;
+    private static ConditionFactory calmlyAwait = Awaitility.with()
+        .pollInterval(slowPacedPollInterval)
+        .and()
+        .with()
+        .pollDelay(slowPacedPollInterval)
```
Comment (rouazana) [inline server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJamesServerTest.java]: maybe you can reduce it :D

Comment (chibenwa) [reply server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJamesServerTest.java]: :D

File: server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJamesServerTest.java [hunk 3]
```diff
@@ -43,7 +43,7 @@
     private static final String PASSWORD = "secret";
     private static final String DOMAIN = "james.org";
     private static final String ADMIN_PASSWORD = "mysecretpassword";
-    private static Duration slowPacedPollInterval = FIVE_HUNDRED_MILLISECONDS;
+    private static Duration slowPacedPollInterval = ONE_HUNDRED_MILLISECONDS;
     private static ConditionFactory calmlyAwait = Awaitility.with()
         .pollInterval(slowPacedPollInterval)
```
Comment (rouazana) [inline server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJamesServerTest.java]: be careful, here you change the pollInterval too.

Comment (chibenwa) [reply server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJamesServerTest.java]: That's wanted. Half a second is too long.

Comment (rouazana) [reply server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJamesServerTest.java]: oops read too fast

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**